adagrams/game.py

```diff
@@ -1,13 +1,129 @@
 from random import randint
 
-def draw_letters():
-    pass
+def draw_letters(): # No parameters
+    
+    # This letter pool should reflect the distribution of letters as described in the Readme file Wave 1
+    distribution_of_letters = { "A": 9, "B": 2, "C": 2, "D": 4, "E": 12, "F": 2, "G": 3, "H": 2, "I": 9, "J": 1, "K": 1, "L": 4, "M": 2, "N": 6, "O": 8, "P": 2, "Q": 1, "R": 6, "S": 4, "T": 6, "U": 4, "V": 2, "W": 2, "X": 1, "Y": 2, "Z": 1 }
+    
+    letter_pool = [] # This represents the pool of letters that the player can draw from
 
-def uses_available_letters(word, letter_bank):
-    pass
+    for letter, frequency in distribution_of_letters.items():
+        for i in range(frequency):
+            letter_pool.append(letter)
+
+
+    # Each string should contain exactly one letter
+    # There are only 2 available C letters, so draw_letters cannot ever return more than 2 Cs
+    # Since there are 12 Es but only 1 Z, it should be 12 times as likely for the user to draw an E as a Z
+    # Invoking this function should not change the pool of letters
+    # Imagine that the user returns their hand to the pool before drawing new letters
+
+    hand = [] # These represent the hand of letters that the player has drawn
+    random_index = 0
+    for i in range(10):
+        random_index = randint(0, len(letter_pool) - 1) # The letters should be randomly drawn from a pool of letters
+        
+
+        while letter_pool[random_index] == 0: # If the randomly drawn letter has already been drawn, then draw again
+            random_index = randint(0, len(letter_pool) - 1)
+        hand.append(letter_pool[random_index])
+        letter_pool[random_index] = 0 # Once a letter is drawn, it cannot be drawn again until the pool is reset
+
+        #  alternative approach without setting to 0:
+        # hand.append(letter_pool.pop(random_index))
+
+
+    return hand # Returns an array of ten strings
+
+def uses_available_letters(word, letter_bank): # Has two parameters:
+    """
+    Checks if the input word can be formed from the letters in the letter bank.
+    
+    Parameters:
+        word (string): describes some input word.
+        letter_bank (list): describes an array of drawn letters in a hand.
+
+    Returns:
+        type: True of False.
+    """
+    # Returns True if every letter in the input word is available (in the right quantities) in the letter_bank
+    # Returns False if not; if there is a letter in input that is not present in the letter_bank or has too much of compared to the letter_bank
+    list_of_available_letters = []
+    for letter in letter_bank:
+        list_of_available_letters.append(letter.upper())
```

Comment:
You can capitalize all the letters in the for loop:
```suggestion
    for letter in letter_bank.upper():
        list_of_available_letters.append(letter)
```
Why we'd prefer to call `upper` on line 52 instead of in the for loop:
-`word.upper()` is called once, up front, rather than repeatedly calling `letter.upper()` inside the loop.
-It conveys intent to other developers because it's clear that you're working with the uppercase version of the word throughout the loop.